File: AL/nadongbin/01-greedy/multiplyOrAdd/multiplyOrAdd.ts
function multiplyOrAdd(numStr: string) {
  const nums = numStr.split('').map((s) => parseInt(s))

  let result = nums[0]
  for (let i = 1; i < nums.length; i++) {
    nums[i] < 2 || result < 1 ? (result += nums[i]) : (result *= nums[i])
  }

  return result
}

console.log(multiplyOrAdd('02984'))

File: AL/nadongbin/01-greedy/multiplyOrAdd/multiplyOrAdd.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func multiplyOrAdd(numStr string) int {
	numChars := strings.Split(numStr, "")
	nums := make([]int, len(numChars))
	for i, nc := range numChars {
		nums[i], _ = strconv.Atoi(nc)
	}
	result := nums[0]
	for i := 1; i < len(nums); i++ {
		next := nums[i]

		if result < 2 || next < 2 {
			result += next
		} else {
			result *= next
		}

		fmt.Println(result)
	}

	return result
}

func main() {
	multiplyOrAdd("12984")
}
